File: CenterElGhlaba/UserIdentity/Hubs/CommentHub.cs
using Center_ElGhalaba.Models;
using Center_ElGhlaba.Interfaces;
using Microsoft.AspNetCore.SignalR;

namespace Center_ElGhlaba.Hubs
{
    public class CommentHub : Hub
    {
        private readonly IUnitOfWork _unitOfWork;
        public CommentHub(IUnitOfWork unitOfWork) 
        {
            _unitOfWork = unitOfWork;
        }
        public async Task AddNewComment(int lessonId, int studentId, string comment, DateTime date)
        {
            LessonComment newComment = new LessonComment()
            {
                LessonID = lessonId,
                StudentID = studentId,
                Body = comment,
                Date = date,
                IsDeleted = false
            };

            _unitOfWork.comments.Insert(newComment);
            _unitOfWork.Complete();

            Student student = await _unitOfWork.Students
                .FindAsync(s => s.ID == studentId, new[] { "AppUser" });
            string username = student.AppUser.FirstName + " " + student.AppUser.LastName;

            await Clients.All.SendAsync("CommentAdded", username, comment, date);
        }
    }
}

File: CenterElGhlaba/UserIdentity/wwwroot/assets/js/commentHub.js
var commentHub = new signalR.HubConnectionBuilder().withUrl("/lessonComment").build();

commentHub.start().then(function () {
    console.log("Comment Connect Success");
});

commentHub.on("CommentAdded", function (username, comment, date) {


    $("#comments-Section").prepend(
        `
            <div class="vid">
                <div>
                    <span class="gray">${username}: </span>
                    <span class="gray">${date}</span>
                    <p>${comment}</p >
                </div>
            </div>
        `
    )
});

function AddComment(lessonID, studentID) {
   
    let comment = $("#comment").val();
    $("#comment").val('');
    
    commentHub.invoke("AddNewComment", lessonID, studentID, comment, new Date());
   
}
